aslam_offline_calibration/kalibr2_ros/test/BagReaderTest.cpp

```diff
@@ -122,4 +123,51 @@ TEST(DISABLED_BagReaderTest, CanDetectMultipleImagesSingleTopic) {
   }
 }
 
+TEST(BagReaderTest, Integration) {
+  std::string bag_path =
+      std::string(TEST_DATA_DIR) + "/rosbag2_2025_06_11-12_00_21_0.mcap";
+  if (!std::filesystem::exists(bag_path)) {
+    GTEST_SKIP() << "Bag file does not exist: " << bag_path;
+  }
+  auto reader =
+      kalibr2::ros::create_bag_reader(bag_path, "/BFS_25037070/image");
+
+  auto target_grid =
+      boost::make_shared<aslam::cameras::GridCalibrationTargetAprilgrid>(
+          5, 6, 0.088, 0.2954);
+  auto geometry =
+      boost::make_shared<kalibr2::models::DistortedPinhole::Geometry>();
+  auto detector = aslam::cameras::GridDetector(geometry, target_grid);
+
+  size_t min_num_observation = 10;
+  std::vector<aslam::cameras::GridCalibrationTargetObservation> observations;
+
+  while (reader->HasNext()) {
+    kalibr2::Image img = reader->ReadNext();
+    assert(!img.image.empty());
+
+    auto observation = kalibr2::ToObservation(img, detector);
+
+    if (observation.has_value()) {
+      observations.push_back(observation.value());
+      std::cout << "Found target in image with timestamp: "
+                << img.timestamp.toNSec() << std::endl;
+      if (observations.size() >= min_num_observation) {
+        std::cout << "Enough observations collected: " << observations.size()
+                  << std::endl;
+        break;  // Stop after collecting enough observations
+      }
+    } else {
+      std::cout << "No target found in image with timestamp: "
+                << img.timestamp.toNSec() << std::endl;
+    }
+  }
+
+  double focal_lenght = 881.0;
```

Comment:
@frneer typo: 
```suggestion
  double focal_length = 881.0;
```

Also, can be constexpr:

```suggestion
  constexpr double focal_length = 881.0;
```

Reply:
Done.